packages/headless/src/modules/brands/repositories/index.ts

```diff
@@ -0,0 +1,35 @@
+import { Product } from '../../../entity/product';
+import { IBrandDatabase } from './brand.database.interface';
+import { Injectable } from '@nestjs/common';
+import { Brand } from 'src/entity/brand';
+
+@Injectable()
+export class BrandRepository {
+    constructor(private readonly db: IBrandDatabase) { }
+
+    async findBrandById(brandId: string): Promise<Brand | null> {
```

Comment:
Do the same thing as suggested before (naming convention)

Reply:
if we need to search anything based on other criterias, then would have to rename the query paramter as well, but for now the api is to find by ID, that is why it is set like that

Reply:
that doesn't matter, because our function returns the same things as well. Different criteria will be handled in the controller layer. our database will not be dependent on any different query parameter or something.